src/main/java/io/vlingo/symbio/store/object/jdbc/JDBCObjectStoreEntryReaderActor.java

```diff
@@ -77,16 +90,45 @@ public void close() {
     return completes().with(name);
   }
 
+  private List<Entry<String>> readIds(List<Long> ids) {
+    try {
+      PreparedStatement statement = queries.statementForEntriesQuery(ids.size());
+      for (int i = 0; i < ids.size(); i++) {
+        // parameter index starts from 1
+        statement.setLong(i + 1, ids.get(i));
+      }
+
+      try (final ResultSet result = statement.executeQuery()) {
+        return mapQueriedEntriesFrom(result);
+      }
+    } catch (Exception e) {
+      logger().info("vlingo/symbio-jdbc: " + getClass().getSimpleName() + " Could not read ids because: " + e.getMessage(), e);
+      return new ArrayList<>();
+    }
+  }
+
   @Override
   public Completes<Entry<String>> readNext() {
     try {
       entryQuery.clearParameters();
       entryQuery.setLong(1, offset);
       try (final ResultSet result = entryQuery.executeQuery()) {
         final Entry<String> entry = mapQueriedEntryFrom(result);
-        ++offset;
-        updateCurrentOffset();
-        return completes().with(entry);
+        List<Long> gapIds = reader().detectGaps(entry, offset);
```

Comment:
@pflueras See the comment in the other primary `readNext(many)`. I would appreciate reusing this code in some shared `JDBCGapResolver`. It will make development of all the other JDBC `EntryReader`'s much quicker to accomplish.

Reply:
This is a good idea! I would suggest to do this refactoring after the implementation of other `EntryReader`s. Then we will have a good overview of all use cases.